### PR TITLE
Create validate_close_amount and add tests for it

### PR DESCRIPTION
**Pull request by maneva3**

### CODE REVIEW

File: contracts/lease/src/position/mod.rs [hunk 1]
```diff
@@ -781,3 +816,94 @@ mod test_check {
...
+
+    use super::Position;
+
+    type TestCurrency = Dai;
+    type TestLpn = Usdc;
+
+    #[test]
+    fn too_small_amount() {
+        let spec = position(100.into(), 75.into(), 15.into());
+        let result = spec.validate_close_amount(14.into(), price(1.into(), 1.into()));
```
Comment (gmanev7) [inline contracts/lease/src/position/mod.rs:841]: [all tests] Using 1:1 price all the time tests just one set of cases. I would recommend replacing it with another easy to compute value, e.g. 2:1

File: contracts/lease/src/position/mod.rs [hunk 2]
```diff
@@ -781,3 +816,94 @@ mod test_check {
...
+    fn valid_amount() {
+        let spec = position(100.into(), 40.into(), 15.into());
+        let result = spec.validate_close_amount(53.into(), price(1.into(), 1.into()));
+        assert!(result.is_ok())
+    }
+
+    fn position(
+        amount: Coin<TestCurrency>,
+        min_asset: Coin<TestLpn>,
+        min_sell_asset: Coin<TestLpn>,
```
Comment (gmanev7) [inline contracts/lease/src/position/mod.rs:885]: introduce type arguments that are `: Into<Currency<Xyz>>` to simplify the calling, e.g. `position(100, 25, 1)`

File: contracts/lease/src/position/status.rs [hunk 3]
```diff
@@ -52,11 +52,13 @@ impl<Asset> Status<Asset>
 where
     Asset: Currency,
 {
-    pub(crate) fn partial(amount: Coin<Asset>, cause: Cause) -> Self {
+    #[cfg(test)]
+    pub fn partial(amount: Coin<Asset>, cause: Cause) -> Self {
```
Comment (gmanev7) [inline contracts/lease/src/position/status.rs:56]: why not `pub(crate)` as the `fn full'` is?

File: contracts/lease/src/position/mod.rs (hunk 1 above)
Comment (maneva3) [reply contracts/lease/src/position/mod.rs]: TODO as a task for PR#177 - Improve tests for check_liability.

File: contracts/lease/src/position/mod.rs [hunk 4]
```diff
@@ -153,16 +154,49 @@ where
         cause: Cause,
         liquidation: Coin<Asset>,
         lpn_in_assets: Price<Lpn, Asset>,
-    ) -> Option<Status<Asset>> {
+    ) -> Option<Liquidation<Asset>> {
+        match self.validate_close_amount(liquidation, lpn_in_assets) {
+            Err(ContractError::PositionCloseAmountTooSmall(_)) => None,
+            Err(ContractError::PositionCloseAmountTooBig(_)) => Some(Liquidation::Full(cause)),
+            Err(_) => unreachable!(),
+            Ok(()) => Some(Liquidation::Partial {
```
Comment (gmanev7) [inline contracts/lease/src/position/mod.rs:162]: please add an assert that the liquidation amount is less than the position amount ...

File: contracts/lease/src/position/mod.rs [hunk 5]
```diff
@@ -153,16 +154,49 @@ where
...
+
+    /// Check if the amount can be used to close the position.
+    /// Return `error::ContractError::PositionCloseAmountTooSmall` when a partial close is requested
+    /// with amount less than the minimum sell asset position parameter sent on lease open. Refer to
+    /// `NewLeaseForm::position_spec`.
+    ///
+    /// Return `error::ContractError::PositionCloseAmountTooBig` when a partial close is requested
+    /// with amount that would decrease a position less than the minimum asset parameter sent on
+    /// lease open. Refer to `NewLeaseForm::position_spec`.
+    pub fn validate_close_amount(
```
Comment (gmanev7) [inline contracts/lease/src/position/mod.rs:177]: please reorder it up to belong to the public functions

File: contracts/lease/src/position/mod.rs [hunk 6]
```diff
@@ -153,16 +154,49 @@ where
...
+            ))
         } else {
-            Some(Status::partial(liquidation, cause))
+            None
+        };
+
+        match may_have_error {
+            Some(error) => Err(error),
+            None => Ok(()),
         }
```
Comment (gmanev7) [inline contracts/lease/src/position/mod.rs:200]: unnecessary complication, just prepare Result above

**Review by gmanev7 — APPROVED** [review]

**Pull request merged**